x-pack/plugins/apm/server/routes/services/route.ts

```diff
@@ -264,13 +265,37 @@ const serviceMetadataDetailsRoute = createApmServerRoute({
       kuery: '',
     });
 
-    return getServiceMetadataDetails({
+    const serviceMetadataDetails = await getServiceMetadataDetails({
       serviceName,
       setup,
       searchAggregatedTransactions,
       start,
       end,
     });
+
+    if (serviceMetadataDetails?.container?.ids) {
+      const {
+        savedObjects: { client: savedObjectsClient },
+        elasticsearch: { client: esClient },
+      } = await context.core;
+
+      const indexName = await getMetricIndices({
+        infraPlugin: plugins.infra,
+        savedObjectsClient,
+      });
+
+      const containerMetadata = await getServiceContainerMetadata({
+        esClient: esClient.asCurrentUser,
+        indexName,
+        containerIds: serviceMetadataDetails.container.ids,
+        start,
+        end,
+      });
+
+      return mergeWith(serviceMetadataDetails, containerMetadata);
+    }
+
+    return serviceMetadataDetails;
```

Comment:
I'd like an API test that also covers the infra metrics data. Is that not possible currently because we can't produce the data? 
Perhaps we can resort to es archives for infra metrics?

Reply:
That's right, we can test only with`es archives` and it seems we already have a mapping for metrics
https://github.com/elastic/kibana/blob/9c90c14b25bb02e05c9bddc8e14cd04e82160f4d/x-pack/test/apm_api_integration/common/fixtures/es_archiver/metrics_8.0.0/mappings.json#L1674-L1766

going to add the api test

Reply:
Yes, I added them to test the infrastructure tab